src/dmd/errors.d

```diff
@@ -443,11 +444,11 @@ private void verrorPrint(const ref Loc loc, Color headerColor, const(char)* head
         // ignore invalid files
         loc != Loc.initial &&
         // ignore mixins for now
-        !loc.filename.strstr(".d-mixin-") &&
+        !loc.filename.ptr.strstr(".d-mixin-") &&
```

Comment:
All these `.ptr` uses assume the string is 0 terminated, which is a fairly dangerous assumption. The use of `strstr` is also wrong. See https://www.tutorialspoint.com/c_standard_library/c_function_strstr.htm

Reply:
The reason I dislike C string handling so much is because every time I review a use of the C string functions, I find a bug in it.